activerecord/lib/active_record/core.rb

```diff
@@ -216,15 +216,26 @@ def self.connection_handler=(handler)
       end
 
       def self.connection_handlers
-        unless legacy_connection_handling
+        if legacy_connection_handling
+        else
           raise NotImplementedError, "The new connection handling does not support accessing multiple connection handlers."
         end
 
         @@connection_handlers ||= {}
       end
 
       def self.connection_handlers=(handlers)
-        unless legacy_connection_handling
+        if legacy_connection_handling
+          ActiveSupport::Deprecation.warn(<<~MSG)
+            Using legacy connection handling is deprecated. Please set
+            `legacy_connection_handling` to `false` in your application.
+
+            The new connection handling does not support `connection_handlers`
+            getter and setter.
+
+            Read more about how to migrate at:
```

Comment:
Read more link is missing. Is it intentionally left blank until guide changes are released?

We can use `https://guides.rubyonrails.org/active_record_multiple_databases.html#migrate-to-the-new-connection-handling` considering after guide release the link would work fine.

Reply:
Whoops, thanks for catching that. I pushed a fix in b8b1c9eba9ade21fd1e48c7d4b8f3b1d46b843fc